### ISSUE TITLE
Pauses during printing multiple labels: Zebra LP2844

### ISSUE THREAD

**Issue by nemolus** [apple/cups#5159]
Hi,

When printing multiple labels, the label printer, Zebra LP2844, will pause for about half a second before printing the next label. 

This significantly slows down the entire printing job if the printer needs to print, say, 100 labels.

Printer is connected to a Linux computer via USB;
Two units of the same model behave exactly the same;
 
However, the same printer using its official driver on a Win7 machine can print continuously without any pauses/stops.

Printer: Zebra LP2844
CUPS version: 2.1.x and 2.2.x
OS: Debian/Arch

Not sure if it's just some checkboxes I didn't tick or could this be a bug?

Thank you very much!

**Comment by michaelrsweet** [comment]
@nemolus Are these labels separate print jobs? This could be an artifact of how the USB backend works on Linux.

**Comment by nemolus** [comment]
Hi @michaelrsweet,

It is one single pring job. Before printing multiple labels, I have to select the "Collate" option in pdf software on Linux/macOS, otherwise, only 1 label will be printed.

As for USB backend on Linux, it could be right! I did a test on a macbook running latest macOS (CUPS 2.2.5), and although there is a pause, the pause is shorter, but not as short as on Win7 where the flow is smooth without noticeable pause.

So to summarize, when the printer is set to be on its fastest mode,
Win7: smooth, no pause
macOS: a short pause (about 0.2-0.3s)
Linux: a longer pause (about 0.5s)

If it is indeed how USB backend works on Linux, is there any settings we can tweak? Thank you!

**Comment by michaelrsweet** [comment]
Hmm, if it is just a single CUPS print job then I don't understand - I have a Zebra GX420d (replacement for the 2844) label printer on my desk and don't see any delays over USB...

What options are you using when you print?

**Comment by nemolus** [comment]
Here are print settings on _evince_, a Linux pdf software:
![Print_Setting_1](https://i.imgur.com/MDzJFIt.png)
![Print_Setting_2](https://i.imgur.com/XcfEEba.png)
![Print_Setting_3](https://i.imgur.com/N28VVI0.png)
![Print_Setting_4](https://i.imgur.com/snOsdFH.png)
![Print_Setting_5](https://i.imgur.com/qfMAfQY.png)

============================================================
Here are system (Arch, CUPS 2.2.6) "Printer Settings" from CUPS web interface:

Current Connection: | usb://Zebra/LP2844?serial=42A062201328
Driver: Zebra EPL2 Label Printer (grayscale)
Defaults: job-sheets=none, none media=oe_w216h72_3x1in sides=one-sided

**General**
Media size: 3x1"
Resolution: 203dpi
Media Type: Printer default (same as Direct Thermal)

**Printer Settings**
Darkness: printer default
Print rate: 6inch / sec

**Banners**
Start: None
End: None

**Error Policy**
Error Policy: retry-current-job
Operation Policy: default

Thanks!

**Comment by michaelrsweet** [comment]
The only difference looks to be that I am using the ZPL driver instead of the EPL driver - can you try the ZPL driver?

**Comment by nemolus** [comment]
ZPL did not work, CUPS reports the job is completed, but printer has no response at all.

It appears LP2844 only takes EPL codes:
https://stackoverflow.com/questions/8497397/zebra-lp2844-stopped-responding-to-zpl-commands-or-files

At the bottom of the printer, its label shows a sqaure with EPL inside, probably suggesting it's an EPL printer.

**Comment by michaelrsweet** [comment]
Hmm, OK well it might be an EPL-specific issue. Will see if I can make my Zebra printer do it with the EPL driver...

**Comment by michaelrsweet** [comment]
Well, I did a test file with 10 3x1" labels on my printer in EPL mode - the first couple labels have a slight delay between them but the rest come out in a continuous stream.

Keep in mind that the CUPS driver only sends raster data while the Windows driver (I think) tries to do text and vector graphics whenever possible, so the size of the label "pages" may be a lot smaller. Could be worth printing to a file on Windows and then doing "lp -dprintername -oraw windows-file.prn" to see if you get the same delays?

**Comment by nemolus** [comment]
Hi @michaelrsweet , you're absolutely right!

I did some tests per your suggestion:

1. On Win7, I added a "Print to File" virtual printer using LP2844 Windows driver;
2. Then I print 3 pages of a pdf file using this printer, saved it as "3.prn"
3. Copied it to Linux home and did "lp -d printer -oraw ~/.3.prn". Voila! These 3 labels came out continuously, no delays, beautiful!
4. Even on Win7, while Acrobat Pro prints continuously, LibraOffice Draw (Windows ver) prints the same pdf with exactly the same pause/delay.

So does this mean it's not a problem of CUPS, but rather how these software handles printing?

If you don't mind, may I ask what package/app/software you used to print your 3x1 label?

**Comment by michaelrsweet** [comment]
I used Pages on my Mac.

Acrobat will draw vectors with the Windows APIs while LibreOffice draws images, so I don't think this is a CUPS issue per-se but just a performance issue of the printer over USB - try printing to a file from LibreOffice and you'll probably see the LibreOffice print file is much larger...

**Comment by michaelrsweet** [comment]
Closing for now; will re-open if you find more information pointing to a necessary software change.
